src/hotspot/share/cds/heapShared.cpp

```diff
@@ -1082,29 +768,26 @@ void KlassSubGraphInfo::add_subgraph_entry_field(int static_field_offset, oop v)
 // Only objects of boot classes can be included in sub-graph.
 void KlassSubGraphInfo::add_subgraph_object_klass(Klass* orig_k) {
   assert(CDSConfig::is_dumping_heap(), "dump time only");
-  Klass* buffered_k = ArchiveBuilder::get_buffered_klass(orig_k);
 
   if (_subgraph_object_klasses == nullptr) {
     _subgraph_object_klasses =
       new (mtClass) GrowableArray<Klass*>(50, mtClass);
   }
 
-  assert(ArchiveBuilder::current()->is_in_buffer_space(buffered_k), "must be a shared class");
-
-  if (_k == buffered_k) {
+  if (_k == orig_k) {
     // Don't add the Klass containing the sub-graph to it's own klass
     // initialization list.
     return;
   }
 
-  if (buffered_k->is_instance_klass()) {
+  if (orig_k->is_instance_klass()) {
+    InstanceKlass* ik = InstanceKlass::cast(orig_k);
```

Comment:
Should this be inside a "#if ASSERT" since `ik` is used only in the `assert` statements?

Reply:
Fixed.